waku/waku_rln_relay/group_manager/static/group_manager.nim

```diff
@@ -107,8 +117,24 @@ else:
 when defined(rln_v2):
   method withdraw*(g: StaticGroupManager, 
                    idSecretHash: IdentitySecretHash): Future[void] {.async: (raises: [Exception]).} =
-    # No-op
-    return
+    initializedGuard(g)
+
+    let groupKeys = g.groupKeys
+
+    for i in 0..<groupKeys.len:
+      if groupKeys[i].idSecretHash == idSecretHash:
+          let idCommitment = groupKeys[i].idCommitment
+          let index = MembershipIndex(i)
+          let rateCommitment = RateCommitment(idCommitment: idCommitment, 
+                                              userMessageLimit: g.userMessageLimit.get())
+          let memberRemoved = g.rlnInstance.removeMember(index)
+          if not memberRemoved:
+            raise newException(ValueError, "Failed to remove member from the merkle tree")
+
+          if g.withdrawCb.isSome():
+            await g.withdrawCb.get()(@[Membership(rateCommitment: rateCommitment, index: index)])
```

Comment:
Not super critical but shall we separate this line into two statements so that it is clear the item that `withdrawCb` is returning?

Reply:
addressed in 249c50ce